pkg/syncer/resourcesync/controller.go

```diff
@@ -261,6 +220,8 @@ func (c *Controller) process(ctx context.Context, key string) error {
 
 	requiredGVRs := getAllGVRs(syncTarget)
 
+	c.cachedDiscovery.Invalidate()
```

Comment:
surprising

Reply:
Rediscovering on every sync target change is very strange.

Reply:
In fact I assume we could invalidate only in the `UpdateFunc()` of the Event Handler, when the `synctarget.Status.SyncedResources` changed between `oldObj` and `newObj` ?